### PR TITLE
fix: amplitude batch output metadata

### PR DESCRIPTION
**Pull request by Gauravudia**
## What are the changes introduced in this PR?
- Converted the metadata in the output to an array for the Amplitude batch endpoint.
- Updated the test cases to replace "job_id" with "jobId" for consistency across all destinations.

## What is the related Linear task?

Resolves INT-1383

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [x] All related docs linked with the PR?

- [x] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [x] Is the type of change in the PR title appropriate as per the changes?

- [x] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3077/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3077/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3077/test-report.html

**Review by chandumlg — APPROVED** [review]

**Review by anantjain45823 — APPROVED** [review]

File: test/integrations/destinations/am/batch/data.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ export const data = [
                 endpoint: 'https://api.eu.amplitude.com/2/httpapi',
               },
               metadata: {
-                job_id: 1,
+                jobId: 1,
```
Comment (krishna2020) [inline test/integrations/destinations/am/batch/data.ts:64]: which one does rudder-server understand - `job_id` or `jobId` ?

**Review by koladilip — APPROVED** [review]

File: test/integrations/destinations/am/batch/data.ts (hunk 1 above)
Comment (Gauravudia) [reply test/integrations/destinations/am/batch/data.ts:64]: jobId

Comment (koladilip) [reply test/integrations/destinations/am/batch/data.ts:64]: https://github.com/rudderlabs/rudder-server/blob/7dc7747e252214c0e0f2a39381ee96c75b4024c7/router/types/types.go

jobId

File: src/v0/destinations/am/transform.js [hunk 2]
```diff
@@ -22,13 +22,13 @@ const {
   getFieldValueFromMessage,
   getValueFromMessage,
   deleteObjectProperty,
-  getErrorRespEvents,
```
Comment (krishna2020) [inline src/v0/destinations/am/transform.js:25]: There are too many other destinations using `getErrorRespEvents` --> So all of those are also impacted ?

**Review by krishna2020 — APPROVED** [review]

File: src/v0/destinations/am/transform.js (hunk 2 above)
Comment (Gauravudia) [reply src/v0/destinations/am/transform.js:25]: Yes

Comment (Gauravudia) [reply src/v0/destinations/am/transform.js:25]: Yes

Comment (krishna2020) [reply src/v0/destinations/am/transform.js:25]: https://linear.app/rudderstack/issue/INT-1579/fix-all-destinations-using-geterrorrespevents-to-get-right-metadata

**Pull request merged**